File: src/main/java/org/example/collection/Color.java
package org.example.collection;

import java.util.LinkedList;
import java.util.Map;

/**
 * Класс перечисление цветов
 */
public enum Color {
    GREEN,
    YELLOW,
    ORANGE,
    BROWN;

    /**
     * @return возвращает все значения перечисления
     */
    public static String getValue() {
        LinkedList<String> enumValues = new LinkedList<>();
        for (Color color : Color.values()) {
            enumValues.add(color.toString());
        }
        return String.join(", ", enumValues);
    }
}

File: common/src/main/kotlin/collection/Color.kt
package collection

import java.io.Serializable
import java.util.*
import java.util.stream.Collectors


enum class Color: Serializable {
    GREEN,
    YELLOW,
    ORANGE,
    BROWN;

    companion object {
        val value: String
            get() = Arrays.stream(entries.toTypedArray())
                .map { obj: Color -> obj.name }
                .collect(Collectors.joining(", "))
    }
}
